content/traceablestorage.go

```diff
@@ -0,0 +1,22 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package content
+
+// TraceableStorage represents a traceable-CAS that supports parent node finding.
+type TraceableStorage interface {
```

Comment:
We probably need a better name other than `TraceableXXX`. Since we have `ExtendedCopy`, how about `ExtendedStorage`?

Reply:
It's an option. But users might be confused when looking at the name `ExtendedStorage` -- extend what? 🤔
How about `SourceableStorage`?

Reply:
Here the `Storage` stands for CAS. It only cares about raw content regardless of what kind of data structure it is storing. In terms of the definition, it is better to be named as `GraphStorage`, isn't it?

Reply:
`GraphStorage` sounds good!